### DISCUSSION TITLE
Init Echo by emitting a livewire event before it is required by listeners.

### DISCUSSION THREAD

**Discussion by vivere-dally** [livewire/livewire#8182] [1. Help]
I had an issue where Echo would be initialized in the boot file and would be available for the entire app. However, I need sockets for only certain pages of the app, and I would like not to waste a connection.

Thus, I tried the following:

```js
window.addEventListener("connect-echo", () => {
  if (!window.Echo) {
    window.Echo = new Echo({
        broadcaster: "reverb",
        key: import.meta.env.VITE_REVERB_APP_KEY,
        wsHost: import.meta.env.VITE_REVERB_HOST,
        wsPort: import.meta.env.VITE_REVERB_PORT ?? 80,
        wssPort: import.meta.env.VITE_REVERB_PORT ?? 443,
        forceTLS: (import.meta.env.VITE_REVERB_SCHEME ?? "https") === "https",
        enabledTransports: ["ws", "wss"],
    });
  }
});
```

And in my component:

```php
class MyComponent extends Component
{
    public function mount(): void
    {
        $this->dispatch("connect-echo");
    }

    /**
     * @param array<string,mixed> $event
     */
    #[On('echo-private:my-channel,MyEvent')]
    public function handleChatMessage(array $event): void
    {
        // do stuff
    }

    public function render(): View|Factory
    {
        return view('livewire.my-component');
    }
}
```

However, I get warnings from Livewire that Echo cannot be found. Thus, I figure that by the time my `connect-echo` event is reaching the window listener, the `On` listener tries to use Echo.

Is there any way to do this in a reliable manner such that Echo would be initialized before any Livewire component actually needs it?

**Comment by joshhanley** [comment]
@vivere-dally Livewire doesn't support dispatching events in mount as Livewire's javascript hasn't loaded yet.

But what you could do, is on pages that you want to fire it, you could just use Alpine or a Livewire javascript hook like `document.addEventListener('livewire:init', () => {...})`

I would use Alpine as it would be the easiest:

```blade
<div x-data x-init="$dispatch('connect-echo')">
```
